cli/azd/cmd/auth_login.go

```diff
@@ -389,6 +389,18 @@ func runningOnCodespacesBrowser(ctx context.Context, commandRunner exec.CommandR
 }
 
 func (la *loginAction) login(ctx context.Context) error {
+	if la.flags.federatedTokenProvider == string(auth.AzurePipelinesFederatedTokenProvider) {
+		if la.flags.clientID == "" {
+			log.Printf("setting client id from environment variable %s", auth.AzurePipelinesClientIDEnvVarName)
+			la.flags.clientID = os.Getenv(auth.AzurePipelinesClientIDEnvVarName)
+		}
+
+		if la.flags.tenantID == "" {
+			log.Printf("setting tenant id from environment variable %s", auth.AzurePipelinesClientIDEnvVarName)
+			la.flags.clientID = os.Getenv(auth.AzurePipelinesTenantIDEnvVarName)
```

Comment:
typo, maybe?

```suggestion
			la.flags.tenantId = os.Getenv(auth.AzurePipelinesTenantIDEnvVarName)
```

Should we fail if env vars are not set?  or is it ok to wait to the log-in error at later ?

Reply:
> typo, maybe?

Yup, fixed!

> Should we fail if env vars are not set? or is it ok to wait to the log-in error at later ?

We could - that's how things work on GitHub (where you must pass both `--client-id` and `--tenant-id` explicitly.  I figured since we could source these from environment variables (since the task sets them) it would be nice to support allowing the user to not specify them when `azure-pipelines` is the `federated-credential-provider` but I'm not wed to that decision.   I do think the ergonomics are a little nicer.